cmd/controller/main.go

```diff
@@ -120,7 +124,7 @@ func main() {
 	flag.IntVar(&queueMaxRetries, "dynamic-controller-default-queue-max-retries", 20,
 		"maximum number of retries for an item in the queue will be retried before being dropped")
 	// log level flags
-	flag.IntVar(&logLevel, "log-level", 10, "The log level verbosity. 0 is the least verbose, 5 is the most verbose.")
+	flag.IntVar(&logLevel, "log-level", 0, "The log level verbosity. Lower values reduce verbosity (e.g., -1=debug, 0=info, 1=warn, 2=error).")
```

Comment:
I think we wouldnt exactly need a custom level enabler. shouldnt controller runtime already bring a log enabler zap logger that follows similar (not same) behavior? PTAL at https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/log/zap#Options.BindFlags and try to adopt the controller runtime logger where possible.

From flag bindings

> zap-log-level: Zap Level to configure the verbosity of logging. Can be one of 'debug', 'info', 'error', 'panic' or any integer value > 0 which corresponds to custom debug levels of increasing verbosity").

for me this is a good behavior we can also use. I would be willing to accept breaking change on the flags even to normalize across other controllers (using controller runtime flags is very common nowadays thanks to kubebuilder)

Reply:
Regarding the Helm chart, I’ll avoid introducing a breaking change by renaming the variables, to maintain compatibility with existing configurations. What do you think @jakobmoellerdev?